quickwit/quickwit-control-plane/src/indexing_scheduler/scheduling/README.md

```diff
@@ -0,0 +1,105 @@
+# Scheduling logic
+
+Quickwit needs to assign indexing tasks to a set of indexers nodes.
+We call the result of this decision the indexing physical plan.
+
+This needs to be done under the constraints of:
+- not exceeding the maximum load of each node. (O)
+
+We also want to observe some interesting properties such as:
+- (A) we want to avoid moving indexing tasks from one indexer to another one needlessly.
+- (B) we want a source to be spread amongst as few nodes as possible
+- (C) we prefer to respect some margin on the capacity of all nodes.
+- (D) when we are working with the Push API source, we prefer to colocate indexers on
+  the ingesters holding the data.
+
+# Problem abstraction
+
+To simplify the logic and make it easier to test it, we first abstract this in the following
+optimization problem. In Quickwit, we have two types of source:
+
+- The push api source: they have a given (changing) set of shards associated to them.
+  A shard is rate-limited to ensure their throughput is lower than `5MB/s` worth of
+  uncompressed data. This guarantees that a given shard can be indexed by a
+  single indexing pipeline.
+
+- Other sources, like Kafka. It is a very common use case to use quickwit to index large
+  amounts of historical data. Right now, the user is therefore expected to supply a desired
+  number of pipeline.
+
+Routers send their batch to the different ingesters they know using a round-robin logic.
+We assume that routers's list of known shards gets eventually updated after a shard addition, so that
+we can assume that shard have roughly the same load.
```

Comment:
Just to be sure to understand the sentence "shard have roughly the same load": if a client sends 43MB/s, we will end up opening 9 shards, each ingesting 43/9= 4.77MB/s right? So yes roughly the same load.

Reply:
Yes.